### PR TITLE
WINDUPRULE-11: provide rule to detect usage of JOTM JTA transactions

### PR DESCRIPTION
**Pull request by m-brophy**
https://issues.jboss.org/browse/WINDUPRULE-11

### CODE REVIEW

File: rules-reviewed/jotm/jotm.rhamt.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This ruleset finds JOTM JTA implementations.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final" />
+        </dependencies>
+        <targetTechnology id="eap" versionRange="[7,8)" />
```
Comment (jeichler) [inline rules-reviewed/jotm/jotm.rhamt.xml]: from 6 onward please

File: rules-reviewed/jotm/jotm.rhamt.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        <rule id="jotm-00001">
+            <when>
+                <javaclass references="org.objectweb.jotm.{*}">
+                    <location>IMPORT</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="JOTM usage" effort="2" category-id="mandatory">
+                    <message>
+                        JOTM JTA implementations are not loaded by EAP on deployment
```
Comment (jeichler) [inline rules-reviewed/jotm/jotm.rhamt.xml]: A bit more details would be good:
your message + the likely error which will occur + the solution + the reference (which you already included)

**Comment by m-brophy** [comment]
Changed to address @jeichler comments

File: rules-reviewed/jotm/jotm.rhamt.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0"?>
```
[the remark this answers is not in the document]
Comment (m-brophy) [reply rules-reviewed/jotm/jotm.rhamt.xml]: done

File: rules-reviewed/jotm/jotm.rhamt.xml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This ruleset finds JOTM JTA implementations.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final" />
+        </dependencies>
+        <targetTechnology id="eap" versionRange="[6,7,8)" />
```
[the remark this answers is not in the document]
Comment (m-brophy) [reply rules-reviewed/jotm/jotm.rhamt.xml]: done

File: rules-reviewed/jotm/tests/jotm.rhamt.test.xml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0"?>
+<ruletest id="jotm-tests"
+          xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <testDataPath>data</testDataPath>
+    <rulePath>../jotm.rhamt.xml</rulePath>
+    <source>false</source>
```
[the remark this answers is not in the document]
Comment (m-brophy) [reply rules-reviewed/jotm/tests/jotm.rhamt.test.xml]: done

File: rules-reviewed/jotm/tests/jotm.rhamt.test.xml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    <ruleset>
+        <rules>
+            <rule id="jotm-00001-test">
+                <when>
+                    <not>
+                        <hint-exists message="JOTM JTA implementations are not loaded"/>
```
[the remark this answers is not in the document]
Comment (m-brophy) [reply rules-reviewed/jotm/tests/jotm.rhamt.test.xml]: done

**Comment by mrizzi** [comment]
Looks good to me, thanks @m-brophy and @jeichler

**Pull request merged**